lib/vanagon/component.rb

```diff
@@ -290,6 +290,13 @@ def get_build_dir
       end
     end
 
+    def get_dependency_hash
+      deps = {}
+      deps['version'] = @version if @version
+      deps['ref'] = @options[:ref] if @options[:ref]
+      { name => deps }
+    end
```

Comment:
Use the accessors, not the instance variables. That's what they're there for. And no need to check first, since the value of those keys in a Hash would be `nil` anyway:

```ruby
{ name => { 'version' => version, 'ref' => ref } }
```

if you _really, **really**_ want to strip out empty keys, try this:

```ruby
{ name => { 'version' => version, 'ref' => ref }.delete_if { |k, v| v.empty? } }
```

(I generally wouldn't worry about whether the keys are in there even if they're empty, since, again, any key in a Hash can be called and will return `nil` if it's not defined or if it's empty.)